CI builds failing with NAN initializer element is not a compile-time constant

Recent GitHub CI builds are failing with this error in libs-corebase:
```
CFNumber.c:129:3: error: initializer element is not a compile-time constant
 |   ( NAN )
 |   ^~~~~~~
```

The Nov 1 2024 build still worked, but on Dec 1 2024 it failed. Since libs-corebase didn’t change it’s probably related to the update of the GitHub Actions [runner updates](https://github.com/actions/runner-images/commits/main/images/windows/Windows2022-Readme.md) on Nov 16 or Nov 26. I’m unable to reproduce locally with the latest Visual Studio 2022 release though.

--- a/phases/40-gnustep-corebase.sh
+++ b/phases/40-gnustep-corebase.sh
@@ -32,7 +32,8 @@ echo "### Running configure"
 
 echo
 echo "### Building"
-make -j`nproc`
+# define _UCRT_NOISY_NAN to work around https://developercommunity.visualstudio.com/t/NAN-is-no-longer-compile-time-constant-i/10688907
+make -j`nproc` CFLAGS=-D_UCRT_NOISY_NAN
 
 echo
 echo "### Installing"

